xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp

```diff
@@ -192,69 +208,142 @@ bool CRendererDRMPRIME::Supports(ESCALINGMETHOD method)
-      if (ret < 0)
-      {
-        CLog::Log(LOGERROR, "CRendererDRMPRIME::%s - failed to retrieve the GEM handle from prime fd %d, ret = %d", __FUNCTION__, descriptor->objects[object].fd, ret);
-        return;
-      }
+      CLog::Log(LOGERROR, "CVideoLayerBridgeDRMPRIME::%s - failed to convert prime fd %d to gem handle %u, ret = %d",
+                __FUNCTION__, descriptor->objects[object].fd, buffer->m_handles[object], ret);
+      return false;
     }
+  }
 
-    AVDRMLayerDescriptor* layer = &descriptor->layers[0];
+  AVDRMLayerDescriptor* layer = &descriptor->layers[0];
 
-    for (int plane = 0; plane < layer->nb_planes; plane++)
+  for (int plane = 0; plane < layer->nb_planes; plane++)
+  {
+    int object = layer->planes[plane].object_index;
+    uint32_t handle = buffer->m_handles[object];
+    if (handle && layer->planes[plane].pitch)
     {
-      int object = layer->planes[plane].object_index;
-      uint32_t handle = buffer->m_handles[object];
-      if (handle && layer->planes[plane].pitch)
-      {
-        handles[plane] = handle;
-        pitches[plane] = layer->planes[plane].pitch;
-        offsets[plane] = layer->planes[plane].offset;
-        modifier[plane] = descriptor->objects[object].format_modifier;
-      }
+      handles[plane] = handle;
+      pitches[plane] = layer->planes[plane].pitch;
+      offsets[plane] = layer->planes[plane].offset;
+      modifier[plane] = descriptor->objects[object].format_modifier;
     }
+  }
 
-    // add the video frame FB
-    ret = drmModeAddFB2WithModifiers(m_DRM->GetFileDescriptor(), buffer->GetWidth(), buffer->GetHeight(), layer->format, handles, pitches, offsets, modifier, &buffer->m_fb_id, 0);
-    if (ret < 0)
+  if (modifier[0] && modifier[0] != DRM_FORMAT_MOD_INVALID)
+    flags = DRM_MODE_FB_MODIFIERS;
+
+  // add the video frame FB
+  ret = drmModeAddFB2WithModifiers(m_DRM->GetFileDescriptor(), buffer->GetWidth(), buffer->GetHeight(), layer->format,
+                                   handles, pitches, offsets, modifier, &buffer->m_fb_id, flags);
+  if (ret < 0)
+  {
+    CLog::Log(LOGERROR, "CVideoLayerBridgeDRMPRIME::%s - failed to add fb %d, ret = %d", __FUNCTION__, buffer->m_fb_id, ret);
+    return false;
+  }
+
+  Acquire(buffer);
+  return true;
+}
+
+void CVideoLayerBridgeDRMPRIME::Unmap(CVideoBufferDRMPRIME* buffer)
+{
+  if (buffer->m_fb_id)
+  {
+    drmModeRmFB(m_DRM->GetFileDescriptor(), buffer->m_fb_id);
+    buffer->m_fb_id = 0;
+  }
+
+  for (int i = 0; i < AV_DRM_MAX_PLANES; i++)
+  {
+    if (buffer->m_handles[i])
     {
-      CLog::Log(LOGERROR, "CRendererDRMPRIME::%s - failed to add drm layer %d, ret = %d", __FUNCTION__, buffer->m_fb_id, ret);
-      return;
+      struct drm_gem_close gem_close = { .handle = buffer->m_handles[i] };
+      drmIoctl(m_DRM->GetFileDescriptor(), DRM_IOCTL_GEM_CLOSE, &gem_close);
+      buffer->m_handles[i] = 0;
     }
+  }
+}
+
+void CVideoLayerBridgeDRMPRIME::Configure(CVideoBufferDRMPRIME* buffer)
```

Comment:
move to header?

Reply:
I left it here and empty in order to make it easer to prepare patches that make use of mainline drm properties, e.g. color encoding/range and [content type](https://patchwork.freedesktop.org/patch/222913/).

Maybe the entire CVideoLayerBridgeDRMPRIME should be moved to VideoLayerBridgeDRMPRIME.cpp/h as it will probably grow in the future ?